.github/workflows/check-branch.yml

```diff
@@ -8,8 +8,19 @@ jobs:
   check_branch:
     runs-on: ubuntu-latest
     steps:
-      - name: Check branch
-        if: github.head_ref != 'development'
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Check if branch is based on development
         run: |
-          echo "ERROR: You can only merge to main from development."
-          exit 1
+          # Fetch development branch
+          git fetch origin development
+
+          # Check if the current branch contains commits from the development branch
+          if git merge-base --is-ancestor origin/development HEAD; then
```

Comment:
I think this is backwards. This will pass if the current branch is ahead of `origin/development` and what you really want is for this to pass when `HEAD` is the same or behind `origin/development`, right? If I have this straight, I think you would want:
```
if git merge-base --is-ancestor HEAD origin/development; then
```

Reply:
Upon testing, I have confirmed that you are correct :)